File: frontend/types/group.ts
import { GroupEvent } from "./event";
import { Post } from "./post";
import { User } from "./user";

export interface Group {
  id: number;
  title: string;
  description: string;
  image: string;
  owner_id: number;
  is_owner: boolean;
  is_accepted: boolean;
  is_pending?: boolean;
  members: User[];
  posts: Post[];
  events: GroupEvent[];
}

File: backend/pkg/model/group.go
package model

import "time"

type Group struct {
	ID           int       `json:"id"`
	OwnerId      int       `json:"owner_id"`
	Members      []*User   `json:"members"`
	Posts      []*Post   `json:"posts"`
	Events      []*GroupEvent   `json:"events"`
	Title        string    `json:"title"`
	Description  string    `json:"description"`
	Image        string    `json:"image"`
	CreationDate time.Time `json:"creation_date"`
	IsOwner      bool      `json:"is_owner"`
	IsAccepted   bool      `json:"is_accepted"`
	IsPending   bool      `json:"is_pending"`
	HasNewEvent     bool   `json:"new_event"`
}
